graph/api/src/main/scala/org/apache/spark/graph/api/CypherSession.scala

```diff
@@ -0,0 +1,186 @@
+    createGraph(nodes.asScala, relationships.asScala)
+  }
+
+  /**
+   * Creates a [[PropertyGraph]] from nodes and relationships.
+   *
+   * The given DataFrames need to adhere to the following column naming conventions:
+   *
+   * {{{
+   *     Id column:        `$ID`            (nodes and relationships)
+   *     SourceId column:  `$SOURCE_ID`     (relationships)
+   *     TargetId column:  `$TARGET_ID`     (relationships)
+   *
+   *     Label columns:    `:{LABEL_NAME}`  (nodes)
+   *     RelType columns:  `:{REL_TYPE}`    (relationships)
+   *
+   *     Property columns: `{Property_Key}` (nodes and relationships)
+   * }}}
+   *
+   * @see [[CypherSession]]
+   * @param nodes         node DataFrame
+   * @param relationships relationship DataFrame
+   * @since 3.0.0
+   */
+  def createGraph(nodes: DataFrame, relationships: DataFrame): PropertyGraph = {
+    val idColumn = CypherSession.ID_COLUMN
+    val sourceIdColumn = CypherSession.SOURCE_ID_COLUMN
+    val targetIdColumn = CypherSession.TARGET_ID_COLUMN
+
+    val labelColumns = nodes.columns.filter(_.startsWith(CypherSession.LABEL_COLUMN_PREFIX)).toSet
+    val nodeProperties = (nodes.columns.toSet - idColumn -- labelColumns)
+      .map(col => col -> col)
+      .toMap
+
+    val labelSets = labelColumns.subsets().toSet + Set.empty
+    val nodeFrames = labelSets.map { labelSet =>
+      val predicate = labelColumns
+        .map {
+          case labelColumn if labelSet.contains(labelColumn) => nodes.col(labelColumn)
+          case labelColumn => !nodes.col(labelColumn)
+        }
+        .reduce(_ && _)
+
+      NodeFrame(nodes.filter(predicate), idColumn, labelSet.map(_.substring(1)), nodeProperties)
+    }
+
+    val relColumns = relationships.columns.toSet
+    val relTypeColumns = relColumns.filter(_.startsWith(CypherSession.LABEL_COLUMN_PREFIX))
+    val propertyColumns = relColumns - idColumn - sourceIdColumn - targetIdColumn -- relTypeColumns
+    val relProperties = propertyColumns.map(col => col -> col).toMap
+    val relFrames = relTypeColumns.map { relTypeColumn =>
+      val predicate = relationships.col(relTypeColumn)
+
+      RelationshipFrame(
+        relationships.filter(predicate),
+        idColumn,
+        sourceIdColumn,
+        targetIdColumn,
+        relTypeColumn.substring(1),
+        relProperties)
+    }
+
+    createGraph(nodeFrames.toSeq, relFrames.toSeq)
+  }
+
+  /**
+   * Loads a [[PropertyGraph]] from the given location.
+   *
+   * @param path directory in which the graph is stored
+   * @since 3.0.0
+   */
+  def load(path: String): PropertyGraph
+
+  /**
+   * Saves a [[PropertyGraph]] to the given location.
+   *
+   * @param graph     [[PropertyGraph]] to be stored
+   * @param path      directory in which the graph should be stored
+   * @param saveMode  specifies what happens when the destination already exists
+   * @since 3.0.0
+   */
+  def save(graph: PropertyGraph, path: String, saveMode: SaveMode): Unit
```

Comment:
This is different from DataFrameWriter pattern. `graph.write.mode("...").save(path)`

Reply:
Do we want to adapt to that pattern? Is this changing with DSv2 in Spark 3.0?

Reply:
Yep. For this one, @mengxr 's advice is the recent common development direction inside Apache Spark. Please follow @mengxr 's advice.
> Do we want to adapt to that pattern? Is this changing with DSv2 in Spark 3.0?

Reply:
We have now done this in c50b679